### PR TITLE
 Allow override for finding record during session creation

### PR DESCRIPTION
**Pull request by avinoth**
* Allow users to custom define the find method for the resource by given email

### CODE REVIEW

**Comment by mikker** [comment]
Looks good to me 👍

Can you add a test case and a note to the Readme?

Thank you!

**Comment by avinoth** [comment]
@mikker Done 👍

File: README.md [hunk 1]
```diff
@@ -118,6 +119,25 @@ app/views/passwordless/mailer/magic_link.text.erb
 
 See [the bundled views](https://github.com/mikker/passwordless/tree/master/app/views/passwordless).
 
+### Overrides
+
+By default `passwordless` uses the `passwordless_with` column you specify in the model to case insensitively fetch the resource during authentication. You can override this and provide your own customer fetcher by defining a class method `fetch_resource_for_passwordless` in your passwordless model. The method will be supplied with the downcased email and should return an `ActiveRecord` instance of the model.
+
+Example time:
```
Comment (mikker) [inline README.md]: Ha!

File: README.md [hunk 2]
```diff
@@ -118,6 +119,25 @@ app/views/passwordless/mailer/magic_link.text.erb
...
+
+Let's say we would like to fetch the record and if it doesn't exist, create automatically.
+
+```ruby
+    class User < ApplicationRecord
```
Comment (mikker) [inline README.md]: Indentation?

File: test/controllers/passwordless/sessions_controller_test.rb [hunk 3]
```diff
@@ -12,6 +12,12 @@ def create_session_for(user)
       )
     end
 
+    def User.fetch_resource_for_passwordless(email)
+      return if email == 'invalidemail'
+
+      User.find_or_create_by(email: email)
+    end
```
Comment (mikker) [inline test/controllers/passwordless/sessions_controller_test.rb]: Hmm, should we undefine it after this spec? This seems like something that could give some pretty weird bugs later in the test suite.

File: README.md (hunk 2 above)
Comment (avinoth) [reply README.md]: Fixed 👍

File: test/controllers/passwordless/sessions_controller_test.rb (hunk 3 above)
Comment (avinoth) [reply test/controllers/passwordless/sessions_controller_test.rb]: Yeah. I've moved the definition inside the respective test, and cleaning it up after

**Comment by mikker** [comment]
👌

**Pull request merged**